nixos/tests/jbake.nix

```diff
@@ -0,0 +1,19 @@
+import ./make-test.nix ({pkgs, ...} : 
+
+{
+  name = "jbake";
+  meta.maintainers = with pkgs.stdenv.lib.maintainers; [ moaxcp ];
+
+  machine = { pkgs, ... }: {
+    environment.systemPackages = [ pkgs.jbake ];
+  };
+
+  testScript = ''
+    $machine->start;
+    $machine->waitForUnit("default.target");
+
+    my $expected = "JBake v${pkgs.jbake.version}";
+    my $result = $machine->succeed("jbake");
```

Comment:
This is not testing a service. We do not have yet the [test derivations](http://nixcon2017.org/#program) as proposed by @Profpatsch.
However what you could add to `development/tools/jbake/default.nix` is:

```nix
{ # ...
checkPhase = ''
  jbake | grep -q "${version}" || (echo "jbake did not return correct version"; exit 1)
'';
doCheck = true;
}
```

Reply:
This will be also seen during build time, while for the test we have to remember that it exists.

Reply:
Thanks! This would probably have been easier. When do nixos/tests get run?

Reply:
When `nixos/release.nix` is build on hydra as part of the nixos jobset.